linera-views/src/backends/lru_caching.rs

```diff
@@ -50,54 +58,267 @@ static NUM_CACHE_SUCCESS: LazyLock<IntCounterVec> = LazyLock::new(|| {
 /// keep track of this.
 
 /// The data structures
+#[allow(clippy::type_complexity)]
 struct LruPrefixCache {
-    map: BTreeMap<Vec<u8>, Option<Vec<u8>>>,
-    queue: LinkedHashMap<Vec<u8>, (), RandomState>,
+    map_find_key_values: BTreeMap<Vec<u8>, Vec<(Vec<u8>, Vec<u8>)>>,
+    map_find_keys: BTreeMap<Vec<u8>, Vec<Vec<u8>>>,
```

Comment:
I'm not sure it's a good idea to cache these, especially as a single entries, since they can contain any number of keys or key-value pairs, so their size is unbounded.

We could not cache the `find_` results at all.

Or we could only store the fact that `find_keys` or `find_key_values` was called with a particular prefix at all, and then cache the actual _entries_ in the other map(s). Whenever an entry belonging to a prefix falls out of the LRU cache or gets deleted, that prefix would have to be cleared here, too.

Reply:
The size argument applies just as well to the `read_value`LRU cache. Sure, the DynamoDb has some limit like 400K, but with the value splitting that limit is passed over.

Reply:
I added a check so that if the cache entry being inserted is larger than the maximal size, then it is refused since that would lead to total clearing of the cache.

Reply:
@MathieuDutSik Let's keep the discussion going and make sure that we go for the best solution here.

Reply:
In my view the current approach is still not ideal, because:
* It potentially duplicates entries, if the same value occurs in both a "find" and a "value" result.
* It creates weird semantics for moving "find" results to the front of a queue based on a "value" cache hit.

I think it would be better to do something like this:

* Whenever we handle a "find" query we put all found entries into the cache as individual "value", and in addition make a note that the cache now contains everything in that range.
* Whenever an entry falls out of the cache we remove (or even better: split or shrink) any such range entry, because now the cache doesn't contain everything under it anymore.
* If we get a "find" query for a range contained in an up-to-date one, we can bump all entries in the _smaller_ range to the front of the queue, i.e. only those that we actually return.